homeassistant/components/danfoss_air/binary_sensor.py

```diff
@@ -32,28 +36,13 @@ class DanfossAirBinarySensor(BinarySensorEntity):
     def __init__(self, data, name, sensor_type, device_class):
         """Initialize the Danfoss Air binary sensor."""
         self._data = data
-        self._name = name
-        self._state = None
+        self._attr_name = name
+        self._attr_is_on = None
```

Comment:
`is_on` cannot be `None`.

Reply:
I have removed the initialisation because it defaults to `None` anyway in the base entity:
https://github.com/home-assistant/core/blob/bcdeb06a4e3bb838531f602ac69087b282b0a9be/homeassistant/components/binary_sensor/__init__.py#L184

I am also not sure why you say that it cannot be `None`:
https://github.com/home-assistant/core/blob/bcdeb06a4e3bb838531f602ac69087b282b0a9be/homeassistant/components/binary_sensor/__init__.py#L196-L199